src/algokit/core/sandbox.py

```diff
@@ -120,6 +120,52 @@ def ps(self, service_name: str | None = None) -> list[dict[str, Any]]:
         assert isinstance(data, list)
         return cast(list[dict[str, Any]], data)
 
+    def _get_local_image_version(self, image_name: str) -> str:
+        """
+        Get the local version of a Docker image
+        """
+        arg = "{{.RepoDigests}}"
+        local_version = run(
+            ["docker", "image", "inspect", image_name, "--format", arg],
+            cwd=self.directory,
+            bad_return_code_error_message="Failed to get image inspect",
+        )
+        # Remove brackets and split on '@' symbol and get the SHA hash
+        return local_version.output[1:-1].split("@")[1]
```

Comment:
output has a newline character at the end which isn't accounted for, can result in a bug where it forgets to strip the rightmost `]` symbol. See if there are better ways to get just the sha part, maybe --format flag has some further ways to parse the output on docker side instead? Otherwise make sure to run strip() on the output before further operations

Reply:
Please also make sure to also reflect the real outputs in the mocks, current tests aren't catching scenarios like that because mocked return values are already trimmed. Potentially adding extra asserts where applicable to ensure the sha strings were correctly parsed and etc (there are no such test checks at the moment)

Reply:
I have changed to use the .Id it seems to return the sha formatted on the way we need already

Reply:
@inaie-makerx I'm sorry you can't really use ID in this case case RepoDigest vs ID are derived in different manners and we rely on repo digest on remote image side. All i meant was that when you run algokit cli (aside from mocks in the tests) this inspect response returns a newline character (happens in current state with ID as well). 

Can you please:
1. Make sure the output is stripper/trimmed from any special characters on left/right side before you return it from `_get_local_image_version` ?
2. Change the ID back to RepoDigest? (see [ref](https://stackoverflow.com/questions/56364643/whats-the-difference-between-a-docker-images-image-id-and-its-digest))

Here is what you can do as an option, docker relies on go under the hood so you can rely on go templating capabilities to do something like this:

```python
    def _get_local_image_version(self, image_name: str) -> str | None:
        """
        Get the local version of a Docker image
        """
        try:
            arg = '{{index (split (index .RepoDigests 0) "@") 1}}'
            local_version = run(
                ["docker", "image", "inspect", image_name, "--format", arg],
                cwd=self.directory,
                bad_return_code_error_message="Failed to get image inspect",
            )
            # Remove brackets and split on '@' symbol and get the SHA hash
            return local_version.output.strip()
        except Exception:
            return None
```